lib/Importer/ONNXIFIModelLoader.cpp

```diff
@@ -20,147 +20,46 @@
-static llvm::Error loadWeight(const onnxTensorDescriptorV1 &in, Tensor *T) {
-  // Only support CPU memory tensors.
-  if (in.memoryType != ONNXIFI_MEMORY_TYPE_CPU) {
-    RETURN_ERR("Only support CPU memory tensors.");
-  }
-
-  std::vector<size_t> dims;
-  for (unsigned i = 0; i < in.dimensions; ++i) {
-    dims.push_back(in.shape[i]);
-  }
-
-  if (in.dataType == ONNXIFI_DATATYPE_FLOAT32) {
-    T->reset(ElemKind::FloatTy, dims);
-
-    auto TH = T->getHandle<>();
-    float *data = (float *)in.buffer;
-    for (size_t i = 0; i < TH.size(); ++i) {
-      TH.raw(i) = data[i];
-    }
-  } else if (in.dataType == ONNXIFI_DATATYPE_UINT64 ||
-             in.dataType == ONNXIFI_DATATYPE_INT64) {
-    const bool inDataSigned = in.dataType == ONNXIFI_DATATYPE_INT64;
-    (void)inDataSigned;
-    T->reset(ElemKind::Int64ITy, dims);
-
-    auto TH = T->getHandle<int64_t>();
-    int64_t *data = (int64_t *)in.buffer;
-    for (size_t i = 0; i < TH.size(); ++i) {
-      RETURN_ERR_IF_NOT(
-          (inDataSigned || data[i] >= 0),
-          "Disallow overflow of loaded UINT64 data into Int64ITy.");
-      TH.raw(i) = data[i];
-    }
-  } else if (in.dataType == ONNXIFI_DATATYPE_INT32) {
-    T->reset(ElemKind::Int32ITy, dims);
-
-    auto TH = T->getHandle<int32_t>();
-    int32_t *data = (int32_t *)in.buffer;
-    for (size_t i = 0; i < TH.size(); ++i) {
-      TH.raw(i) = data[i];
-    }
-  } else {
-    RETURN_ERR("Only float and index tensors are supported.");
-  }
-
-  return llvm::Error::success();
-}
-
-llvm::Error ONNXIFIModelLoader::loadWeights(
-    uint32_t weightsCount, const onnxTensorDescriptorV1 *weightDescriptors) {
-  for (uint32_t i = 0; i < weightsCount; ++i) {
-    std::unique_ptr<Tensor> T(new Tensor());
-    RETURN_IF_ERR(loadWeight(weightDescriptors[i], T.get()));
-    tensors_[weightDescriptors[i].name] = std::move(T);
-  }
-
-  return llvm::Error::success();
-}
-
-llvm::Expected<std::unique_ptr<ONNXIFIModelLoader>>
-ONNXIFIModelLoader::parse(const void *onnxModel, uint32_t onnxModelSize,
-                          uint32_t weightsCount,
-                          const onnxTensorDescriptorV1 *weightDescriptors,
-                          Function &F, bool loadInputsAsPlaceholders) {
+llvm::Expected<std::unique_ptr<ONNXIFIModelLoader>> ONNXIFIModelLoader::parse(
+    const void *onnxModel, uint32_t onnxModelSize, uint32_t weightsCount,
+    const onnxTensorDescriptorV1 *weightDescriptors, Function &F,
+    bool loadInputsAsPlaceholders, bool use_onnx) {
   llvm::Error loaderConstructionErr = llvm::Error::success();
   std::unique_ptr<ONNXIFIModelLoader> loader(
-      new ONNXIFIModelLoader(F, &loaderConstructionErr));
+      new ONNXIFIModelLoader(&loaderConstructionErr));
   if (loaderConstructionErr) {
     return std::move(loaderConstructionErr);
   }
 
-  ONNX_NAMESPACE::ModelProto modelDef;
-  ASSIGN_VALUE_OR_RETURN_ERR(modelDef,
-                             loader->loadProto(onnxModel, onnxModelSize));
+  if (use_onnx) {
```

Comment:
So right now if !use_onnx, this method kinda does nothing, right?  Should we assert(use_onnx) for the time being (just until the c2 implementation is in place)?

Reply:
C2 implementation is coming tomorrow, or tonight. :)